### PR TITLE
remove annotation for 3.X release

### PR DESCRIPTION
**Pull request by toxicity1985**

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ jobs:
           - php-version: '7.4'
           - php-version: '8.0'
```
Comment (dbu) [inline .github/workflows/php.yml]: symfony 6.4 needs php 8.1 or newer, we should remove the older builds

File: composer.json [hunk 2]
```diff
@@ -21,11 +21,11 @@
         }
     ],
     "require": {
-        "php": "^7.3 || ^8.0",
+        "php": "^8.2",
```
Comment (dbu) [inline composer.json]: lets do ^8.1 we don't need the very latest and symfony 6.1 allows 8.1

File: composer.json [hunk 3]
```diff
@@ -35,20 +35,19 @@
...
         "mockery/mockery": "^1.3.2",
         "monolog/monolog": "*",
-        "sensio/framework-extra-bundle": "^4.0 || ^5.5.1 || ^6.0",
         "doctrine/annotations": "^1.11",
-        "symfony/browser-kit": "^4.4 || ^5.0 || ^6.0",
-        "symfony/console": "^4.4 || ^5.0 || ^6.0",
-        "symfony/finder": "^4.4 || ^5.0 || ^6.0",
+        "symfony/browser-kit": "^6.0 || ^7.0",
+        "symfony/console": "^6.0 || ^7.0",
+        "symfony/finder": "^6.0 || ^7.0",
```
Comment (dbu) [inline composer.json]: lets make the dev symfony dependencies 6.4 as well, not 6.0

File: src/Configuration/ConfigurationInterface.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+interface ConfigurationInterface
+{
+    /**
+     * Returns the alias name for an annotated configuration.
+     *
+     * @return string
+     */
+    public function getAliasName();
```
Comment (dbu) [inline src/Configuration/ConfigurationInterface.php]:
```suggestion
    public function getAliasName(): string;
```

we now have modern php, so we can use return type declarations. (and remove the redundant `@return string` annotation)

File: src/Http/ResponseMatcher/ExpressionResponseMatcher.php [hunk 5]
```diff
@@ -11,7 +11,7 @@
 
 namespace FOS\HttpCacheBundle\Http\ResponseMatcher;
 
-use Sensio\Bundle\FrameworkExtraBundle\Security\ExpressionLanguage as SecurityExpressionLanguage;
+use FOS\HttpCacheBundle\Security\ExpressionLanguage as SecurityExpressionLanguage;
```
Comment (dbu) [inline src/Http/ResponseMatcher/ExpressionResponseMatcher.php]: do we still need this class or can we remove it in favor of https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/HttpFoundation/RequestMatcher/ExpressionRequestMatcher.php

File: tests/Functional/EventListener/InvalidationListenerTest.php [hunk 6]
```diff
@@ -19,83 +19,6 @@ class InvalidationListenerTest extends WebTestCase
 {
     use MockeryPHPUnitIntegration;
 
-    public function testInvalidateRoute()
```
Comment (dbu) [inline tests/Functional/EventListener/InvalidationListenerTest.php]: do we have corresponding tests with attributes? if not we should convert this to test attributes so we have functional tests of the thing working in action.

**Review by dbu — COMMENTED** [review]
thanks a lot!

i added some comments.

File: .github/workflows/php.yml (hunk 1 above)
Comment (dbu) [reply .github/workflows/php.yml]: but lets keep lowest builds, just with the lowest supported php version

File: composer.json (hunk 2 above)
Comment (toxicity1985) [reply composer.json]: yeah but i switch to symfony 7 and symfony 7 needs php8.2 no ?

Comment (dbu) [reply composer.json]: we allow symfony 6.4 || 7, so this is fine.

composer will make sure that symfony 7 is not installed when on php 8.1, but its maybe helpful for people who for the moment are still on php 8.1 / symfony 6.4.

Comment (toxicity1985) [reply composer.json]: Ok. I didn't know that. Thanks.

File: tests/Functional/EventListener/InvalidationListenerTest.php (hunk 6 above)
Comment (toxicity1985) [reply tests/Functional/EventListener/InvalidationListenerTest.php]: Yes. The test alreay exist on attribute.

**Comment by toxicity1985** [comment]
There is 2 bundle to be updated to be comptabile with php8.2.

**Comment by toxicity1985** [comment]
Do you have an idea why the rule matcher is not ok ?
Something has change but i don't found what.

File: composer.json [hunk 7]
```diff
@@ -21,11 +21,11 @@
         }
     ],
     "require": {
-        "php": "^7.3 || ^8.0",
-        "friendsofsymfony/http-cache": "^2.15",
-        "symfony/framework-bundle": "^4.4.0 || ^5.0 || ^6.0",
-        "symfony/http-foundation": "^4.4.0 || ^5.0 || ^6.0",
-        "symfony/http-kernel": "^4.4.0 || ^5.0 || ^6.0"
+        "php": ">= 8.1",
```
Comment (dbu) [inline composer.json]:
```suggestion
        "php": "^8.1",
```

i know symfony itself does this, but i strongly disagree. major PHP version changes can do BC breaks, so we should not claim compatibility with all future versions of PHP.

File: composer.json [hunk 8]
```diff
@@ -35,21 +35,20 @@
         "guzzlehttp/guzzle": "^7.2",
         "mockery/mockery": "^1.3.2",
         "monolog/monolog": "*",
-        "sensio/framework-extra-bundle": "^4.0 || ^5.5.1 || ^6.0",
         "doctrine/annotations": "^1.11",
```
Comment (dbu) [inline composer.json]: do we still need this?

File: src/Http/RequestMatcher/QuerystringRequestMatcher.php [hunk 9]
```diff
@@ -12,18 +12,23 @@
...
  */
-class QuerystringRequestMatcher extends SymfonyRequestMatcher
+class QuerystringRequestMatcher implements RequestMatcherInterface
 {
     /**
      * @var string Regular expression to match the query string part of the request url
      */
     private $queryString;
 
+    public function __construct($queryString = null)
```
Comment (dbu) [inline src/Http/RequestMatcher/QuerystringRequestMatcher.php]: when we add this to the constructor, i'd like to remove the setter for queryString.

File: composer.json [hunk 10]
```diff
@@ -35,21 +35,20 @@
...
-        "symfony/security-bundle": "^4.4 || ^5.0 || ^6.0",
-        "symfony/twig-bundle": "^4.4 || ^5.0 || ^6.0",
-        "twig/twig": "^2.13",
-        "symfony/yaml": "^4.4 || ^5.0 || ^6.0",
-        "symfony/css-selector": "^4.4 || ^5.0 || ^6.0",
-        "symfony/expression-language": "^4.4 || ^5.0 || ^6.0",
+        "symfony/browser-kit": "^6.4 || ^7.0",
+        "symfony/console": "^6.4 || ^7.0",
+        "symfony/finder": "^6.4 || ^7.0",
+        "symfony/phpunit-bridge": "^7.0",
```
Comment (dbu) [inline composer.json]: we could try forcing the latest patch version of the bridge, maybe that helps:

```suggestion
        "symfony/phpunit-bridge": "^7.0.1",
```

File: src/Configuration/ConfigurationInterface.php [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * ConfigurationInterface.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+interface ConfigurationInterface
+{
+    /**
+     * Returns the alias name for an annotated configuration.
```
Comment (dbu) [inline src/Configuration/ConfigurationInterface.php]: we should talk about attribute, not annotation.

File: src/Configuration/ConfigurationInterface.php [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+namespace FOS\HttpCacheBundle\Configuration;
+
+/**
+ * ConfigurationInterface.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+interface ConfigurationInterface
```
Comment (dbu) [inline src/Configuration/ConfigurationInterface.php]: looking at how we use this in Php8AttributesListener i would prefer to do a `match` on $instance to see if it is one of the 3 known attribute classes and if so add it with the corresponding alias name. this interface is not really adding value anymore i think.

File: src/Command/BaseInvalidateCommand.php [hunk 13]
```diff
@@ -24,8 +23,6 @@
  */
 abstract class BaseInvalidateCommand extends Command
 {
-    use ContainerAwareTrait;
```
Comment (dbu) [inline src/Command/BaseInvalidateCommand.php]: do we need to add a setContainer method and call setContainer in our services?

the recommended way in the upgrade instructions is to not rely on the container anymore - would that be possible here, can we figure out what our code does with the container and inject the specific things to the specific commands?

(that would be out of scope for this MR though. if commands still work with fetching the container from the application object we can merge like this for now)

**Review by dbu — COMMENTED** [review]
thanks for the work!

i try to [get the ball rolling](https://github.com/Jean-Beru/fos-http-cache-cloudfront/pull/3) with the cloudfront extension. worst case we disable the test if the component is not updated for the new version soonish.

File: tests/Functional/EventListener/InvalidationListenerTest.php (hunk 6 above)
Comment (dbu) [reply tests/Functional/EventListener/InvalidationListenerTest.php]: great!

File: src/Command/BaseInvalidateCommand.php (hunk 13 above)
Comment (dbu) [reply src/Command/BaseInvalidateCommand.php]: i created #604 for this

**Comment by toxicity1985** [comment]
Hi,

I update the plugin this morning. I add a match function as asked but the get alias is necessary to add configuration inside the request. I juste have a question regards the invalidation path attribute. If we have a match we have to clear the path configured inside the attribute but not the route called ?

**Pull request closed without merging**